languages/OCaml/romanize.ml

```diff
@@ -0,0 +1,30 @@
+open Batteries
+
+let conversions = [(1,    "I" );
+                   (4,    "IV");
+                   (5,    "V" );
+                   (9,    "IX");
+                   (10,   "X" );
+                   (40,   "XL");
+                   (50,   "L" );
+                   (90,   "XC");
+                   (100,  "C" );
+                   (400,  "CD");
+                   (500,  "D" );
+                   (900,  "CM");
+                   (1000, "M" )]
+
+(* Return the highest Arabic/Roman pair less than or equal to number *)
+let highest_contained arabic = conversions
+                               |> List.filter ((>=) arabic % fst)
+                               |> List.last
+
+(* Recursively convert the Arabic number to a Roman numeral *)
+let rec romanize arabic = match arabic with
+  | 0 -> ""
+  | n -> let (a, r) = highest_contained n in r ^ romanize (n - a);;
```

Comment:
I can't find a coherent explanation for why you need the double semicolons, but you do. Something about the separation between simply defining things and code to be immediately executed. I may just be doing it wrong.